common/configuration/db_config_provider.go

```diff
@@ -0,0 +1,56 @@
+package configuration
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	"github.com/alecthomas/types/optional"
+)
+
+// DBConfigProvider is a configuration provider that stores configuration in its key.
+type DBConfigProvider struct {
+	DB  bool `help:"Write configuration values to the database." group:"Provider:" xor:"configwriter"`
```

Comment:
I think we won't need to integrate with the CLI, so you can probably omit this.

Reply:
awesome!